openbas-model/src/main/java/io/openbas/database/model/Filters.java

```diff
@@ -29,6 +33,31 @@ public static class FilterGroup {
     private FilterMode mode; // Between filters
     private List<Filter> filters;
 
+    // -- UTILS --
+
+    public Optional<Filter> findByKey(@NotBlank final String filterKey) {
+      if (this.getFilters() == null) {
+        return Optional.empty();
+      }
+
+      return this.getFilters()
+          .stream()
+          .filter(filter -> filter.getKey().equals(filterKey))
+          .findFirst();
+    }
+
+    public void removeByKey(@NotBlank final String filterKey) {
+      if (this.getFilters() == null) {
+        return;
+      }
+
+      List<Filter> newFilters = this.getFilters()
+          .stream()
+          .filter(filter -> !filter.getKey().equals(filterKey))
+          .toList();
```

Comment:
filter.getKey() or filterKey can be null?

Reply:
@Data
  public static class Filter {

    @NotNull
    private String key;
    private FilterMode mode; // Between values: name = name1 OR name = name2
    private List<String> values;
    private FilterOperator operator;
  }
  
  filter.getKey() can't be null
  
  @NotBlank final String filterKey
  filterKey can't be null